logging/api/Program.cs

```diff
@@ -0,0 +1,237 @@
+
+        public LoggingSample(TextWriter stdout)
+        {
+            _out = stdout;
+        }
+
+        readonly TextWriter _out;
+
+        public bool PrintUsage()
+        {
+            _out.WriteLine(s_usage);
+            return true;
+        }
+
+        public static int Main(string[] args)
+        {
+            LoggingSample loggingSample = new LoggingSample(Console.Out);
+            return loggingSample.Run(args);
+        }
+
+        // [START write_log_entry]
+        private void WriteLogEntry(string logId, string message)
+        {
+            var client = LoggingServiceV2Client.Create();
+            string logName = $"projects/{s_projectId}/logs/{logId}";
+            LogEntry logEntry = new LogEntry();
+            logEntry.LogName = logName;
+            logEntry.Severity = LogSeverity.Info;
+            Type myType = typeof(LoggingSample);
+            string entrySeverity = logEntry.Severity.ToString().ToUpper();
+            logEntry.TextPayload =
+                $"{entrySeverity} {myType.Namespace}.LoggingSample - {message}";
+            // Set the resource type to control which GCP resource the log entry belongs to.
+            // See the list of resource types at:
+            // https://cloud.google.com/logging/docs/api/v2/resource-list
+            // This sample uses 'global' which will cause log entries to appear in the 
+            // "Global" resource list of the Developers Console Logs Viewer:
+            //  https://console.cloud.google.com/logs/viewer
+            MonitoredResource resource = new MonitoredResource();
+            resource.Type = "global";
+            // Create dictionary object to add custom labels to the log entry.
+            IDictionary<string, string> entryLabels = new Dictionary<string, string>();
+            entryLabels.Add("size", "large");
+            entryLabels.Add("color", "red");
+            // Add log entry to collection for writing. Multiple log entries can be added.
+            IEnumerable<LogEntry> logEntries = new LogEntry[] { logEntry };
+            client.WriteLogEntries(logName, resource, entryLabels, logEntries);
+            _out.WriteLine($"Created log entry in log-id: {logId}.");
+        }
+        // [END write_log_entry]
+
+        // [START list_log_entries]
+        private void ListLogEntries(string logId)
+        {
+            var client = LoggingServiceV2Client.Create();
+            string logName = $"projects/{s_projectId}/logs/{logId}";
+            IEnumerable<string> projectIds = new string[] { s_projectId };
+            var results = client.ListLogEntries(projectIds, logName, "timestamp desc");
+            foreach (var row in results)
+            {
+                if (row != null && !String.IsNullOrEmpty(row.TextPayload.Trim()))
+                {
+                    _out.WriteLine($"{row.TextPayload.Trim()}");
+                }
+                else
+                {
+                    results.GetEnumerator().Dispose();

```

Comment:
I downloaded the code and tried running it, and it blocks forever even with the work-around.  :-(